api/v1alpha1/ratelimitfilter_types.go

```diff
@@ -99,6 +99,23 @@ type RateLimitSelectCondition struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	Headers []HeaderMatch `json:"headers,omitempty"`
+
+	// SourceIP defines the match attributes for source IP addresses from “X-Forwarded-For“ header.
+	// +optional
+	SourceIP *SourceIPMatch `json:"sourceIP,omitempty"`
+}
+
+// SourceIPMatch defines the match attributes for source IP addresses from “X-Forwarded-For“ header.
+type SourceIPMatch struct {
```

Comment:
prefer if this was a `string` , internally the system should be able to validate a IPv4/IPv6 CIDR using something like https://pkg.go.dev/net#ParseCIDR